### PR TITLE
Abhimanyu/epoch 16

### PR DESCRIPTION
**Pull request by Abhimanyu121**

### CODE REVIEW

File: packages/sdk/src/HttpRpcClient.ts [hunk 1]
```diff
@@ -91,6 +95,19 @@ export class HttpRpcClient {
...
+  /**
+   * get userOperations for a user.
+   * @param sender
+   * @return sender - address of smart contract wallet.
+   */
+  async getAdvancedUserOperations(
+    sender: string
+  ): Promise<Array<AdvancedUserOperationListStruct>> {
+    await this.initializing;
+    const resp: Array<AdvancedUserOperationListStruct> =
```
Comment (Man-Jain) [inline packages/sdk/src/HttpRpcClient.ts:101]: SHouldn't this be `AdvancedUserOperationStructList`

File: packages/sdk/src/HttpRpcClient.ts [hunk 2]
```diff
@@ -19,6 +19,10 @@ export class DeleteOperationStruct {
   nonce!: number;
   signature!: string;
 }
+export class AdvancedUserOperationListStruct {
+  chainId!: string;
+  userOp!: UserOperationStruct;
```
Comment (Man-Jain) [inline packages/sdk/src/HttpRpcClient.ts]: Remove the ! operator

**Review by Man-Jain — CHANGES_REQUESTED** [review]

File: packages/sdk/src/HttpRpcClient.ts [hunk 3]
```diff
@@ -13,11 +13,13 @@ import Debug from "debug";
 import { AdvancedUserOperationStruct } from "./AdvancedUserOp";
 
 const debug = Debug("aa.rpc");
-export class DeleteOperationStruct {
-  sender!: string;
-  chainId!: number;
-  nonce!: number;
-  signature!: string;
+export class AdvancedUserOperationListStruct {
```
Comment (Man-Jain) [inline packages/sdk/src/HttpRpcClient.ts]: Change to interface no need for a class

File: packages/sdk/src/HttpRpcClient.ts (hunk 1 above)
Comment (Abhimanyu121) [reply packages/sdk/src/HttpRpcClient.ts:101]: Nope, the data that's coming is an array of this struct.
```[
  {
    chainId: '0x13881',
    userOp: {
      sender: '0x278160b87c275d453FE5d65BaE5001a06799cF6f',
      nonce: '0x0',
      initCode: '0x4a4fc0bf39d191b5fca7d4868c9f742b342a39c15fbfb9cf000000000000000000000000f39fd6e51aad88f6f4ce6ab8827279cfffb922660000000000000000000000000000000000000000000000000000000000000000',
      callData: '0xb61d27f600000000000000000000000031d31065280287de4060c277748884637f533310000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000024db55ebc8000000000000000000000000000000000000000000000000000000000000003200000000000000000000000000000000000000000000000000000000',
      callGasLimit: '0x16e4a0',
      verificationGasLimit: '0x5fa1f',
      preVerificationGas: '0xb0a4',
      maxFeePerGas: '0x59682f20',
      maxPriorityFeePerGas: '0x59682f00',
      paymasterAndData: '0x',
      signature: '0x85d2edcc7c74104ebf7fc14ccb9ccdedba9d529c3f1890dd641fee66ed516aa97c9842f895e8364cc9e46f1668d79ec3d3e60c3bf804a03a7a021e3a766023511b',
      advancedUserOperation: [Object]
    },
    entryPoint: '0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789',
    prefund: '0x0',
    userOpHash: '0x6784a21e9378fb10930f88d0f6c6577e5fd3b448d44e0357a5f67b6630a4e25b',
    lastUpdatedTime: '0x18cda8a431e'
  }
]```

**Review by Man-Jain — APPROVED** [review]

**Pull request merged**